File: TFTP/Client/src/CommandData.cpp
#include "CommandData.h"

CommandData::CommandData(short packetSize, short blockNumber, char* data) : Command((short)3), packetSize(packetSize), blockNumber(blockNumber), data(data){}

CommandData::CommandData(const CommandData& other):Command((short)3), packetSize(other.packetSize), blockNumber(other.blockNumber), data(other.data){}

CommandData& CommandData::operator=(const CommandData& other){
	if (this!=&other){
		packetSize = other.packetSize;
		blockNumber = other.blockNumber;
		char* temp = other.data;
		delete[] data;
		data = temp;
	}
	return *this;
}

short CommandData::getPacketSize(){
	return packetSize;
}

short CommandData::getBlockNumber(){
	return blockNumber;
}

char* CommandData::getData(){
	return data;
}

CommandData::~CommandData(){
	delete[] data;
}

File: TFTP/Server/src/main/java/impl/tftp/CommandData.java
package bgu.spl171.net.impl.tftp;

public class CommandData extends Command {
	
	private short packetSize;
	private short blockNumber;
	byte[] data;
	
	public CommandData(short packetSize, short blockNumber, byte[] data) {
		super((short) 3);
		this.packetSize = packetSize;
		this.blockNumber = blockNumber;
		this.data = data;
	}

	public short getSize() {
		return packetSize;
	}

	public short getBlockNumber() {
		return blockNumber;
	}

	public byte[] getData() {
		return data;
	}
	

}
